bin/ansible-connection

```diff
@@ -36,181 +19,120 @@ import socket
-
-            return pid
-        except OSError as e:
-            sys.exit(1)
-    except OSError as e:
-        sys.exit(1)
-
-
-class Server():
-
-    def __init__(self, socket_path, play_context):
-        self.socket_path = socket_path
+    def __init__(self, fd, play_context, socket_path):
         self.play_context = play_context
+        self.socket_path = socket_path
 
-        display.display(
-            'creating new control socket for host %s:%s as user %s' %
-            (play_context.remote_addr, play_context.port, play_context.remote_user),
-            log_only=True
-        )
-
-        display.display('control socket path is %s' % socket_path, log_only=True)
-        display.display('current working directory is %s' % os.getcwd(), log_only=True)
-
-        self._start_time = datetime.datetime.now()
-
-        display.display("using connection plugin %s" % self.play_context.connection, log_only=True)
-
-        self.connection = connection_loader.get(play_context.connection, play_context, sys.stdin)
-        self.connection._connect()
-
-        if not self.connection.connected:
-            raise AnsibleConnectionFailure('unable to connect to remote host %s' % self._play_context.remote_addr)
+        self.fd = fd
+        self.exception = None
 
-        connection_time = datetime.datetime.now() - self._start_time
-        display.display('connection established to %s in %s' % (play_context.remote_addr, connection_time), log_only=True)
+        self.srv = JsonRpcServer()
+        self.sock = None
 
-        self.socket = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
-        self.socket.bind(self.socket_path)
-        self.socket.listen(1)
-        display.display('local socket is set to listening', log_only=True)
+    def start(self):
+        try:
+            messages = list()
+            result = {}
+
+            messages.append('control socket path is %s' % self.socket_path)
+
+            self.connection = connection_loader.get(self.play_context.connection, self.play_context, '/dev/null')
+            self.connection._connect()
+            self.srv.register(self.connection)
+            messages.append('connection to remote device started successfully')
+
+            self.sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+            self.sock.bind(self.socket_path)
+            self.sock.listen(1)
+            messages.append('local domain socket listeners started successfully')
+        except Exception as exc:
+            result['error'] = to_text(exc)
+            result['exception'] = traceback.format_exc()
+        finally:
+            result['messages'] = messages
+            self.fd.write(json.dumps(result))
+            self.fd.close()
 
     def run(self):
         try:
             while True:
-                signal.signal(signal.SIGALRM, self.connect_timeout)
-                signal.signal(signal.SIGTERM, self.handler)
+                signal.signal(signal.SIGALRM, self.handler)
```

Comment:
Having a separate handler for each timer helps to debug which timer is expired based on debugging logs. With this change is there a way to identify which connection timer is expired?